cep-XXXX.md

```diff
@@ -0,0 +1,216 @@
+
+#### Channel base URLs
+
+The base URL for the arbitrary location of a repodata file is defined as:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]/<subdir>/repodata.json
+```
+
+with `<scheme>`, `<authority>` and `<path>` defined by [RFC
+3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2).
+
+Taken the channel definition above, the base URL without trailing slashes is thus:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]
+```
+
+For example, given `https://conda.anaconda.org/conda-forge/noarch/repodata.json`, the part leading
+to `noarch/repodata.json` and thus base URL is `https://conda.anaconda.org/conda-forge`. For
+local repodata such as `file:///home/username/channel/noarch/repodata.json`, the
+channel base URL is `file:///home/username/channel`.
+
+When present, each path component MUST only contain lowercase ASCII letters, numbers, underscores,
+periods, and dashes. They MUST NOT start with a period or a dash. They SHOULD start and end with a
+letter or a number. If present, each path component MUST match this regex:
+
+```re
+^[a-z0-9_][a-z0-9_.-]*$
+```
+
+For `file://`-based channel URLs, the path component rules MAY be understood as recommendations only.
+
+The maximum length of an individual path component in a channel base URL MUST NOT exceed 128 characters.
+The maximum length of a channel base URL SHOULD NOT exceed 256 characters.
+
+#### Channel names
+
+For convenience, the channel _name_ is defined as the concatenation of `scheme`, `authority` and
+`path` components of a channel URL. At least one of `authority` or `path` SHOULD be present. In
+their absence, the channel name MUST be considered empty, regardless the scheme. Empty channel
+names SHOULD NOT be used.
+
+When the scheme and authority fields are missing, the full URL can be inferred with these rules:
+
+- If the channel name matches the regex `^\.{0,2}[/\\].*$`, or if it matches the regex `^[A-Z]:([\\/].*)?$` (for Windows drives), it SHOULD be understood as the path component of a `file://` URL.
+- Otherwise, it SHOULD be understood as a `http[s]://` URL. The tool SHOULD assume a default scheme and authority (e.g. `https://conda.anaconda.org`), and take the rest as a path component.
+
+#### Subdir names
+
+Channel subdir names MUST either be the literal `noarch` or a string following the syntax `{os}-{arch}`,
+where `{os}` and `{arch}` MUST only consist of lowercase ASCII letters and numbers. Non-`noarch` subdirs
+MUST match this regex: `^[a-z0-9]+-[a-z0-9]+$`.
+
+The maximum length of a subdir name MUST NOT exceed 32 characters.
+
+#### Label names
+
+Channel label names MUST only consist of ASCII letters, digits, underscores, hyphens, forward slashes,
+periods, and whitespace. They MUST start with a letter. They MUST match this regex:
+`^[a-zA-Z][0-9a-zA-Z_\-\.\/\s]*`. Even if allowed, label names SHOULD NOT contain any whitespace.
```

Comment:
@chenghlee @jaimergp as long as we're deviating from anaconda.org here (with the removal of `:`s), we might as well remove whitespace `\s` as well?

Reply:
I'm ok with that.

Reply:
I can also add back `:` and include it in the SHOULD NOT part.

Reply:
I think both `:` and `\s` are bad ideas so I am happy removing them both.

Reply:
Farewell to \s then!